zpool.cpp

```diff
@@ -0,0 +1,687 @@
+
+#include <iostream>
+#include <vector>
+#include <cstring>
+#include <map>
+#include <algorithm>
+#include "zpool.h"
+
+using namespace std;
+	
+map<int,string> Zpool::zpool_list()
+{
+       	int iCount = -1;
+       	FILE * pipe = NULL;
+       	char buffer[128];
+       	map<int, string> poolmap;
+
+
+       	//pipe = popen("echo Ajay@123 | sudo -S  zpool list\n", "r");
+       	pipe = popen("sudo zpool list\n", "r");
```

Comment:
Have you thought of using any other options instead of popen for invoking a command? For simplicity and readability, third party library could have been used.